### PR TITLE
Validate quiz

### PR DESCRIPTION
**Pull request by rlchoy**
BTW i didn't know how to implement the timer functionality properly and i left some code from when i branched off scores + timer so... just ignore most of the other changes in game.ts and gameTypes.ts that don't fall in these categories bc the timer stuff will be fixed by jess:

1. export-quiz endpoint in game.ts: returns quiz json
2. modified score function to use quiz's points instead of a flat score
3. wrote quiz validation

feel free to give me any suggestions

### CODE REVIEW

**Review by celdond — COMMENTED** [review]
I am going to look at the failing tests and see why they are failing.

**Comment by celdond** [comment]
I noticed this code does not take into account an injection, say if there is a field added that should not be there.
i.e. meta has all of the correct keys, except it has an additional key "kenobi".
I have created a few tests, and they do indeed validate despite injections.

File: backend/src/__tests__/quizCreation.test.ts [hunk 1]
```diff
@@ -36,6 +42,39 @@ const wrongMetaTypes = {
...
+
+const injectedList = correct.questions;
+injectedList.push(injectedQuestion);
+
+const questionInjectionQuiz = {
+	meta: correct.meta,
+	questions: injectedList,
+	injection: 'I have the high ground!',
+};
+
```
Comment (celdond) [inline backend/src/__tests__/quizCreation.test.ts:76]: Added these tests for ease of testing injection.

**Review by celdond — CHANGES_REQUESTED** [review]
Validation should take into account keys that should not be there, I provided tests to code against.

File: backend/src/__tests__/quizCreation.test.ts (hunk 1 above)
Comment (rlchoy) [reply backend/src/__tests__/quizCreation.test.ts:76]: you're right, i forgot about that

Comment (celdond) [reply backend/src/__tests__/quizCreation.test.ts:76]: It's alright, I didn't include these tests from the start.
Everything else is great from what I see, good work!

**Review by celdond — APPROVED** [review]

File: backend/src/game.ts [hunk 2]
```diff
@@ -21,6 +21,8 @@ function endQuestion(gameId: GameId) {
 	let qAns = game.quizData.questions[game.activeQuestion].correctAnswers;
 	let leaderboard: { name: string; score: number }[] = [];
 	users.forEach(function (playerId: UserId) {
```
Comment (blahajjessie) [inline backend/src/game.ts:23]: Try to use the functions in `gameTypes.ts`

**Review by blahajjessie — CHANGES_REQUESTED** [review]
Fix timer

**Comment by blahajjessie** [comment]
To be merged in #35

**Pull request closed without merging**